Cython/Build/Inline.py

```diff
@@ -32,20 +32,19 @@ def to_unicode(s):
             return s.decode('ascii')
         else:
             return s
+
+    import imp
+    load_dynamic = imp.load_dynamic
+
 else:
     to_unicode = lambda x: x
 
+    import importlib.util
+    from importlib.machinery import ExtensionFileLoader
 
-if sys.version_info < (3, 5):
-    import imp
-    def load_dynamic(name, module_path):
-        return imp.load_dynamic(name, module_path)
-else:
-    import importlib.util as _importlib_util
-    def load_dynamic(name, module_path):
-        spec = _importlib_util.spec_from_file_location(name, module_path)
-        module = _importlib_util.module_from_spec(spec)
-        # sys.modules[name] = module
+    def load_dynamic(name, path):
+        spec = importlib.util.spec_from_file_location(name, loader=ExtensionFileLoader(name, path))
```

Comment:
I have adjusted the implementation in a way to limit importing only Extension modules.